### DISCUSSION TITLE
Suppress warnings for 3rd party directives

### DISCUSSION THREAD

**Discussion by mason3k** [executablebooks/MyST-Parser#766] [Q&A]
I am looking for the recommended way to resolve a build warning when the directive comes from a separate 3rd-party extension.

Even when using the [only directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-only) to house the third-party extension, I still get the warning for, for example, when running the html builder with a directive meant for the confluence builder:
` WARNING: Unknown directive type: 'confluence_toc' [myst.directive_unknown]`

I know that I could suppress the warning in `conf.py` but is there a way to handle these spurious errors better without suppressing real errors?

**Comment by mason3k** [comment]
I assume `only` doesn't work because it is mostly concerned with filtering the output in a "post" hook:
https://github.com/sphinx-doc/sphinx/issues/1420

**Reply by chrisjsewell** [reply]
Heya, yeh exactly everything in the `only` is still parsed, but just filtered out in the outputs.

I would ask why is the `confluence_toc` directive not loaded?
All 3rd-party libraries should still be loaded, irrespective of what your output is.

Otherwise, you could always create a "dummy" directive in your `conf.py`
